### ISSUE TITLE
Option to disable backspace, delete, navigation keys, and mouse selection to prevent editing while writing

### ISSUE THREAD

**Issue by xahodo** [vkbo/novelWriter#1721]
The basic idea is to have a menu entry or option somewhere which, when enabled, disables all navigation (except from enter) and deletion keys, and puts the cursor automatically at the end of the current text. It would also be useful to disable selection using the mouse while this feature is active. This is to prevents writers from editing the text they've already typed. Of course those keys would only be disabled in the text editor where the actual chapters/scenes are written, not throughout the entire application. It could be called "Disable editing".

Such a tool is really helpful to prevent writers from editing what was entered while they're writing (I suffer from this). This option would be particularly useful in a first draft situation.

Further options building on this feature could be to include options such as automatically turning this off when the user switches to another chapter/scene which already contains text (thought being: the user now wants to edit) and a clock which, once set, would prevent this option from being turned off until the clock reaches zero. It could be a countdown or a clock which is active between certain times.

I tried searching, but could not find a feature request for this.

**Comment by vkbo** [comment]
Yes, I've seen this referred to as both "Typewriter Mode" and "Hemingway Mode". I've thought about adding this already, but you're right that there is no feature request for it.

There could be both a timer option and a word count option to the feature as well.

**Comment by vkbo** [comment]
I was thinking where to add this feature on the GUI. I'm planning to add collapsible panels below the Project Tree, like in VSCode. There are a few features I want to put in there. This may be a good place for this too.

A panel could include a progress bar with a word or time goal, with the option to keep "write only" mode on for the duration. There have been other requests for a writing session progress bar, and I may as well combine them into one.

The writing and editing mode toggle can be located on the editor footer for manual toggle and to display which mode the editor is currently in.

**Comment by vkbo** [comment]
Implementation note: The mode can be implemented by cancelling keystrokes where the cursor is not at the end of the document. This will allow the usage of the backspace key, but not editing anywhere inside the text already written.

Optionally, when the user types, the cursor could also skip to the end. This may be useful if the document ends in line breaks and someone is trying to continue the last paragraph. In which case, nothing would happen as it's not technically the end.

**Comment by xahodo** [comment]
Hasn't Qt6 got some option to filter out _all_ unwanted keystrokes? Having backspace still available might be a bit problematic, as this would still allow for some editing - and I _will_ use that backspace key (discipline problems here).

Either "Typewriter mode" or "Hemmingway mode" as a name for the menu entry would be fine.

It could in the beginning be a simple check-mark, which would turn this feature on or off at the user's discretion. Later the options to have it tied to a timer or word count could be added.

**Comment by xahodo** [comment]
Perhaps it's better to call this "typwriter mode". "Hemmingway mode" would refer to an American author not everybody may know and people could be puzzled by what it means.

**Comment by vkbo** [comment]
"Typewriter" is already used for a different feature.

**Comment by xahodo** [comment]
Why not just drop the wordcount or timer options to this and instead make it a simple toggle?

I've been experimenting with ghostwriter and the toggle is more than adequate. Besides, what happens when you're done with your text and you haven't reached the wordcount or allotted time yet?

**Comment by vkbo** [comment]
> Why not just drop the wordcount or timer options to this and instead make it a simple toggle?

That's a different feature, #2223, but I would like to combine them in the same panel.

**Comment by HeyMyian** [comment]
An optional setting for this feature could be "invisible text", to further encourage the writer to simply keep writing. This could be achieved by setting most or all colours in the [Syntax] section to the same (editor background) colour.

I could have sworn this was mentioned before somewhere, but I can't find anything.
